lib/Dancer2/Serializer/Mutable.pm

```diff
@@ -63,47 +76,53 @@ sub serialize {
 sub deserialize {
     my ( $self, $content ) = @_;
 
-    # The right content type should already be set
-    my $format = $formats->{$self->content_type};
-
-    $format and return $serializer->{$format}{'from'}->( $self, $content );
+    my $format = $self->_get_content_type('content_type');
+    $format and return $serializer->{$format}{'from'}->($self, $content);
 
     return $content;
 }
 
 sub _get_content_type {
-    my $self    = shift;
-    $self->has_request or return;
-
-    # Search for the first HTTP header variable which
-    # specifies supported content.
-    foreach my $method ( qw<content_type accept> ) {
-        if ( my $value = $self->request->header($method) ) {
-            if ( exists $formats->{$value} ) {
-                $self->set_content_type($value);
-                return $formats->{$value};
+    my ($self, $header) = @_;
+
+    if ( $self->has_request ) {
+        # Search for the first HTTP header variable which specifies
+        # supported content. Both content_type and accept are checked
+        # for backwards compatibility.
+        foreach my $method ( $header, qw<content_type accept> ) {
+            if ( my $value = $self->request->header($method) ) {
+                if ( my $serializer = $self->mapping->{$value} ) {
```

Comment:
Should we compare case-insensitively here, just for Postel's sake, in case someone said e.g. `Content-Type: text/HTML` or similar?  (They shouldn't, but it's nice to just DYWM sometimes).